internal/iam/query.go

```diff
@@ -206,6 +206,102 @@ const (
         on grants.role_id = roles.role_id;
     `
 
+	grantsForUserGlobalResourcesQuery = `
+	with
+    users (id) as (
+      select public_id
+        from iam_user
+      %s -- anonUser || authUser
+    ),
+    user_groups (id) as (
+      select group_id
+        from iam_group_member_user
+       where member_id in (select id from users)
+    ),
+    user_accounts (id) as (
+      select public_id
+        from auth_account
+       where iam_user_id in (select id from users)
+    ),
+    user_oidc_managed_groups (id) as (
+      select managed_group_id
+        from auth_oidc_managed_group_member_account
+       where member_id in (select id from user_accounts)
+    ),
+    user_ldap_managed_groups (id) as (
+      select managed_group_id
+        from auth_ldap_managed_group_member_account
+       where member_id in (select id from user_accounts)
+    ),
+    managed_group_roles (role_id) as (
+      select distinct role_id
+        from iam_managed_group_role
+       where principal_id in (select id from user_oidc_managed_groups)
+          or principal_id in (select id from user_ldap_managed_groups)
+    ),
+    group_roles (role_id) as (
+      select role_id
+        from iam_group_role
+       where principal_id in (select id from user_groups)
+    ),
+    user_roles (role_id) as (
+      select role_id
+        from iam_user_role
+       where principal_id in (select id from users)
+    ),
+    user_group_roles (role_id) as (
+      select role_id
+        from group_roles
+      union
+      select role_id
+        from user_roles
+      union
+      select role_id
+        from managed_group_roles
+    ),
+    roles_with_grants (role_id, canonical_grant) as (
+      select iam_role_grant.role_id,
+        iam_role_grant.canonical_grant
+      from iam_role_grant
+      join iam_role
+        on iam_role.public_id               = iam_role_grant.role_id
+      join iam_grant
+        on iam_grant.canonical_grant        = iam_role_grant.canonical_grant
+      where iam_role.public_id in (select role_id from user_group_roles)
+        and iam_grant.resource              = any('{"%s", "unknown", "*"}')
```

Comment:
Likewise this could be done with a parameter, ie:

```sql
and iam_grant.resource = any(@resources)
```

And a named arg:

```go
sql.Named("resources", fmt.Sprintf("{'%s', 'unknown', '*'}", res.String()))